api/src/failpoint.rs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#[allow(unused_imports)]
+use anyhow::{format_err, Result};
+use diem_api_types::Error;
+
+#[allow(unused_variables)]
+#[inline]
+pub fn fail_point(name: &str) -> Result<(), Error> {
+    Ok(fail::fail_point!(format!("api::{}", name).as_str(), |_| {
+        Err(format_err!("unexpected internal error for {}", name).into())
```

Comment:
Worth highlighting the purpose of this function is to help `fail_point!` know the type of error to return in the presence of failure. Documented here: https://docs.rs/fail/latest/fail/#early-return

Reply:
are you asking for adding document?

Reply:
are you asking for adding document?